widget/entry.go

```diff
@@ -1218,14 +1230,15 @@ func (e *Entry) selectingKeyHandler(key *fyne.KeyEvent) bool {
 	switch key.Name {
 	case fyne.KeyBackspace, fyne.KeyDelete:
 		// clears the selection -- return handled
-		e.setFieldsAndRefresh(e.eraseSelection)
+		e.setFields(e.eraseSelection)
 		e.propertyLock.RLock()
 		cb := e.OnChanged
 		e.propertyLock.RUnlock()
 		if cb != nil {
 			cb(e.Text)
 		}
-		e.Validate()
+		e.validate()
+		e.Refresh()
```

Comment:
You have move the refresh to before the callback in many other places. I suppose we want to be as consistent as possible. Not sure what is best but refreshing before potentially calling a long-running callback seems sensible?

Reply:
Yes. You are right.

Reply:
But the validate() won't call the refresh. If so, we need to also move the validate. Is this consistent with the previous behavior? I found in the original code, sometimes we call the Validate() before the callback function ,but sometimes after it.

Reply:
I think it makes more sensor to validate before? The user might want to check the validation status maybe?

Reply:
I honestly don't know. We should be consistent though